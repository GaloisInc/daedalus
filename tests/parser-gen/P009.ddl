{- Tests with Many:
   - SetUnion construct
   - Calling Class function NCCall
-}

def $letter = 'A'..'K' | 'a'..'k' | 'r'..'z'

def Main = { x = Many $letter ; END }
