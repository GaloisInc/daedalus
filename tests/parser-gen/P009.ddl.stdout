module P009
 
--- Imports:
 
--- Type defs:
type P009.Main = { x: [uint 8]
                 }
 
--- Rules:
 
P009.$letter : ByteClass =
  (('A' .. 'K') | ('a' .. 'k')) | ('r' .. 'z')
 
P009.Main : Grammar P009.Main =
  do (x : [uint 8]) <- Many[] Match P009.$letter
     END
     pure {x = x}
 
P009.Main_ : Grammar {} =
  do @Many[] @Match P009.$letter
     END
